### ISSUE TITLE
Difference vs. real file system using Dir.glob("./*") when in root ("/") directory

### ISSUE THREAD

**Issue by andrewdavidmackenzie** [fakefs/fakefs#339]
I have code that looks like this:

> Dir.glob("#{filename}/*").each do |folder_entry|

That seems to have a difference from the real FS.

I run in rspec using FakeFS, clearing the FS, creating two files, and then passing filename = '.'
the array entries from Dir.glob are of the type:

> "/file1"

If I try to recreate in irb:

> cd /
> irb
> 2.1.5 :001 > Dir.glob("./*")
>  => ["./Applications", "./bin", "./cores", "./dev", "./etc", "./home", "./installer.failurerequests", "./Library", "./net", "./Network", "./private", "./sbin", "./System", "./tmp", "./User Information", "./Users", "./usr", "./var", "./Volumes"] 
> 2.1.5 :002 > Dir.pwd
>  => "/" 

I get as I expected, "./file" instead of "/file" as returned by fakefs.

I'll try and create a failing test for this...but wanted to see if you had comments before I went too far.

**Comment by grosser** [comment]
sounds like a obvious bug ... not sure if the fix is obvious too ... but give it a try :D

**Comment by andrewdavidmackenzie** [comment]
def test_dir_glob_keeps_prepended_dot
    Dir.chdir('/')
    assert_equal Dir.glob('./*', 0), ['./tmp']
  end

  1) Failure:
FakeFSTest#test_dir_glob_keeps_prepended_dot [/Users/andrew/workspace/fakefs/test/fakefs_test.rb:1158]:
Expected: ["/tmp"]
  Actual: ["./tmp"]

**Comment by andrewdavidmackenzie** [comment]
in find

> def find(path)
>       parts = path_parts(normalize_path(path))
>       return fs if parts.empty? # '/'
> 
>       entries = Globber.expand(path).flat_map do |pattern|
>         parts = path_parts(normalize_path(pattern))
>         find_recurser(fs, parts).flatten
>       end
> 
>       case entries.length
>       when 0 then nil
>       when 1 then entries.first
>       else entries
>       end
> end

it calls find_recurser with dir = '/' and parts = ['*']

which uses this code:    

>  Globber.expand(pattern).flat_map do |subpattern|
>                     dir.matches(Globber.regexp(subpattern))
>                   end

and so the leading '.' is lost of the directory

**Comment by grosser** [comment]
yeah it's tricky ... I'm no expert either, just merging what looks good / has decent tests :D 
maybe avoid this whole mess and use `Dir.mktmpdir { |d| Dir.chdir(d) { run tests } }`
